### ISSUE TITLE
pgbadger fails to detect log_line_prefix

### ISSUE THREAD

**Issue by bitfag** [dalibo/pgbadger#337]
Hi! Tnx for the awesome tool. We're successfully used pgbadger until December 13, 2016. It suddenly stopped to discover queries in the log. I performed some investigation and found that it works only if --prefix explicitly specified.

The log_line_prefix we're using is from pgbadger README which says it should be autodetected.
```
# grep log_line_prefix /etc/postgresql/9.4/main/postgresql.conf 
log_line_prefix = '%t [%p]: [%l-1] db=%d,user=%u,app=%a,client=%h '
```

```
# /usr/bin/pgbadger /var/log/postgresql/postgresql-9.4-main.log -O /var/lib/postgresql/pg_reports/gits -d gits -v
DEBUG: Autodetected log format 'stderr' from /var/log/postgresql/postgresql-9.4-main.log
DEBUG: Output 'html' reports will be written to out.html
DEBUG: Starting to parse log file: /var/log/postgresql/postgresql-9.4-main.log
DEBUG: Starting reading file /var/log/postgresql/postgresql-9.4-main.log...
DEBUG: Start parsing at offset 0 of file /var/log/postgresql/postgresql-9.4-main.log to 8293457674
[========================>] Parsed 8301997064 bytes of 8293457674 (100.10%), queries: 0, events: 0
...
```

With manually specified prefix it works:

```
# /usr/bin/pgbadger -I --prefix '%t [%p]: [%l-1] db=%d,user=%u,app=%a,client=%h ' /var/log/postgresql/postgresql-9.4-main.log -O /var/lib/postgresql/pg_reports/gits -d gits -v
DEBUG: Autodetected log format 'stderr' from /var/log/postgresql/postgresql-9.4-main.log
DEBUG: Output 'html' reports will be written to out.html
DEBUG: Starting to parse log file: /var/log/postgresql/postgresql-9.4-main.log
DEBUG: This file should be parsed from the beginning: /var/log/postgresql/postgresql-9.4-main.log, reach the end of check_file_changed() with start date: 2016-12-13 15:44:44 and file size: 8387973629
DEBUG: Reverting start offset 7870151485 to 0 for file /var/log/postgresql/postgresql-9.4-main.log, stoppping offset is 8387973629
DEBUG: Starting reading file /var/log/postgresql/postgresql-9.4-main.log...
DEBUG: Start parsing at offset 0 of file /var/log/postgresql/postgresql-9.4-main.log to 8387973629
[========================>] Parsed 8495120078 bytes of 8387973629 (101.28%), queries: 101993, events: 329622
DEBUG: the log statistics gathering took:1836 wallclock secs (1823.49 usr + 11.75 sys = 1835.24 CPU)
```

Log example:

```
# head -2 /var/log/postgresql/postgresql-9.4-main.log
2017-03-05 02:44:54 +05 [4371]: [1-1] db=[unknown],user=[unknown],app=[unknown],client=192.168.66.152 LOG:  connection received: host=192.168.66.152 port=37704
2017-03-05 02:44:54 +05 [23783]: [803-1] db=gits,user=gits,app=[unknown],client=192.168.66.155 LOG:  duration: 1804.042 ms  statement:
```

I wonder could it be the datetime format change due to updated `tzdata` package?

**Comment by darold** [comment]
Hi,

It should work without specifying a prefix. Please post the content of the file: `/var/lib/postgresql/pg_reports/gits/LAST_PARSED`

And what is your version of pgbadger?

**Comment by bitfag** [comment]
> It should work without specifying a prefix. Please post the content of the file: /var/lib/postgresql/pg_reports/gits/LAST_PARSED
>
> And what is your version of pgbadger?


```
# cat /var/lib/postgresql/pg_reports/gits/LAST_PARSED
2017-03-07 17:43:17     8495120078      2017-03-07 17:43:17 +05 [23205]: [3-1] db=gits,user=gits,app=[unknown],client=192.168.66.156 LOG:  disconnection: session time: 0:00:00.075 user=gits database=gits host=192.168.66.156 port=58698
```

```
pgBadger version 8.1
```

**Comment by darold** [comment]
Everything looks fine unless the pgBadger version, please upgrade to 9.1. You can just override your pgbadger installation, it should be fully backward compatible with your current binary data file.

**Comment by bitfag** [comment]
10.03.2017, 13:40, "Darold" <notifications@github.com>:
> Everything looks fine unless the pgBadger version, please upgrade to 9.1. You can just override your pgbadger installation, it should be fully backward compatible with your current binary data file.


Thanks! I updated pgbadger to the latest available version in pgdg repo - 9.0-1.pgdg14.04+1 and it working now:

```
# /usr/bin/pgbadger -I /var/log/postgresql/postgresql-9.4-main.log -O /var/lib/postgresql/pg_reports/gits -d gits -v                                                                                     DEBUG: Autodetected log format 'stderr' from /var/log/postgresql/postgresql-9.4-main.log
DEBUG: Output 'html' reports will be written to out.html
DEBUG: Autodetected log format 'stderr' from /var/log/postgresql/postgresql-9.4-main.log
DEBUG: Autodetected log format 'stderr' from /var/log/postgresql/postgresql-9.4-main.log
DEBUG: Starting to parse log file: /var/log/postgresql/postgresql-9.4-main.log
DEBUG: This might not be parsed: /var/log/postgresql/postgresql-9.4-main.log, timestamp 2017-03-05 02:44:54 read at offset 0 is lower than saved timestamp: 2017-03-07 17:43:17
DEBUG: Starting reading file /var/log/postgresql/postgresql-9.4-main.log...
DEBUG: Start parsing at offset 8495120078 of file /var/log/postgresql/postgresql-9.4-main.log to 17049575317
DEBUG: Unknown line format:                 (
DEBUG: Unknown line format:                     SELECT DISTINCT cp.id, cp.name, cp.code_number, cp.description, ST_X(cp.cp_point::geometry) AS lon, ST_Y(cp.cp_point::geometry) AS lat
DEBUG: Unknown line format:                     FROM route_cached_checkpoint AS rcc
DEBUG: Unknown line format:                     JOIN route AS r ON r.id = rcc.route_id
DEBUG: Unknown line format:                     JOIN checkpoint AS cp ON cp.id = rcc.checkpoint_id
DEBUG: Unknown line format:                     WHERE (rcc.date = NULL OR NULL is NULL) AND r.city_id = 1 AND cp.name iLIKE '%%'
DEBUG: Unknown line format:
DEBUG: Unknown line format:                     ORDER BY cp.name
DEBUG: Unknown line format:                 ) UNION ALL (
DEBUG: Unknown line format:                     SELECT DISTINCT cp.id, cp.name, cp.code_number, cp.description, ST_X(cp.cp_point::geometry) AS lon, ST_Y(cp.cp_point::geometry) AS lat
DEBUG: Unknown line format:                     FROM route_cached_checkpoint AS rcc
DEBUG: Unknown line format:                     JOIN route AS r ON r.id = rcc.route_id
DEBUG: Unknown line format:                     JOIN checkpoint AS cp ON cp.id = rcc.checkpoint_id
DEBUG: Unknown line format:                     WHERE (rcc.date = NULL OR NULL is NULL) AND r.city_id = 1 AND cp.description iLIKE '%%' AND cp.name NOT iLIKE '%%'
DEBUG: Unknown line format:
DEBUG: Unknown line format:                     ORDER BY cp.name
DEBUG: Unknown line format:                 )
DEBUG: Unknown line format:
[============>            ] Parsed  9062609319 bytes of 17049575317 (53.15%), queries: 14943, events: 735
```

**Comment by darold** [comment]
Good news.
